src/internal_events/redis.rs

```diff
@@ -0,0 +1,91 @@
+use super::InternalEvent;
+use metrics::counter;
+
+#[derive(Debug)]
+pub struct RedisEventReceived {
+    pub byte_size: usize,
+}
+
+impl InternalEvent for RedisEventReceived {
+    fn emit_logs(&self) {
+        trace!(message = "Received one event.", rate_limit_secs = 10);
+    }
+
+    fn emit_metrics(&self) {
+        counter!("processed_events_total", 1);
+        counter!("processed_bytes_total", self.byte_size as u64);
+    }
+}
+
+#[derive(Debug)]
+pub struct RedisReceiveEventFailed {
+    pub error: redis::RedisError,
+}
+
+impl InternalEvent for RedisReceiveEventFailed {
+    fn emit_logs(&self) {
+        error!(
+            message = "Failed to read message.",
+            error = %self.error,
+            rate_limit_secs = 30,
+        );
+    }
+
+    fn emit_metrics(&self) {
+        counter!("receive_event_errors_total", 1);
+    }
+}
+
+#[derive(Debug)]
+pub struct RedisEventSent {
+    pub byte_size: usize,
+}
+
+impl InternalEvent for RedisEventSent {
+    fn emit_logs(&self) {
+        trace!(message = "Processed one event.", rate_limit_secs = 10);
+    }
+
+    fn emit_metrics(&self) {
+        counter!("processed_events_total", 1);
```

Comment:
Could we add an analogous `counter!("events_out_total", 1);` here? We've started transitioning to that metric name instead of `processed_events_total`. We should leave `processed_events_total` too though, as the deprecated name.